### ISSUE TITLE
Allow an 'insert new records only' checkbox option

### ISSUE THREAD

**Issue by fjorba** [inveniosoftware/invenio-oaiharvester#40]
Some remote OAI servers provide low quality records that are improved in the local Invenio repository.  Please create a checkbox option to choose if external records can overwrite local records, or just allow importing new records.

This should be a per-OAI-source option, not a general one.

**Comment by kaplun** [comment]
Hi @fjorba. Note that invenio-oaiharvester just implement the harvesting part of Invenio, i.e. to fetch records from external sources. What then is happening to such record is delegated to your specific instance configuration by registering corresponding receivers that listen to the invenio-oaiharvester signals defined in: https://github.com/inveniosoftware/invenio-oaiharvester/blob/ef2fa6ad9be84fbe9bd531d5c80df43adbd4e6f6/invenio_oaiharvester/signals.py#L26

**Comment by kaplun** [comment]
E.g. in INSPIRE these are connected with: https://github.com/inspirehep/inspire-crawler/blob/ceac84364721f997466330c4d833befc56f5750a/inspire_crawler/receivers.py#L38-L59 and it's then the responsibility of the receiver to decide what to do with the record.

**Comment by fjorba** [comment]
Hi @kaplun.  I understand the idea, and it looks perfectly reasonable.  I cannot, however, understand this Python code, that goes beyond my current understanding.  I hope that there will be some librarian-friendly user interface to deal with that case.

For the current 1.1.x release, my solution has been similar to your answer: I've patched bibupload so that, when run with -i (insert only new records), it just ignores existing records and does not give an error.

But we understand that there must be a friendly user-interface option.

**Comment by kaplun** [comment]
Yeah, indeed I see your point. I believe then this feature request should rather belong to the Invenio demosite, since it would be just one of the many proposals about what to do with harvested records.

**Comment by kaplun** [comment]
@invenio-developers what is the best place for this request?
